src/framework/input/element-input.js

```diff
@@ -736,15 +736,23 @@ class ElementInput {
         if (eventType === 'mouseup' && this._pressedElement) {
             // click event
             if (this._pressedElement === this._hoveredElement) {
-                this._pressedElement = null;
-
-                // fire click event if it hasn't been fired already by the touchup handler
-                if (!this._clickedEntities || !this._clickedEntities[this._hoveredElement.entity.getGuid()]) {
+                // fire click event if it hasn't been fired already by the touchend handler
+                const guid = this._hoveredElement.entity.getGuid();
+                // Always fire, if there are no clicked entities
+                let fireClick = !this._clickedEntities;
+                // But if there are, we need to check how long ago touchend added a "click brake"
+                if (this._clickedEntities) {
+                    const lastTouchUp = this._clickedEntities[guid] || 0;
+                    const dt = Date.now() - lastTouchUp;
+                    fireClick = dt > 300;
+                }
+                // We do not check another time, so the worst thing that can happen is one ignored click in 300ms.
```

Comment:
Is 300ms dependent on device speed/performance?

Reply:
Yea, that is the question, I tested on my devices and its either around 50ms or 270ms (newish iPhone vs oldish Android). In Chrome Touch devtools, I got 250ms (Chrome on Linux). Considering extreme lag / bad performance (lets say `mouseup` would be delayed to 400ms), then a double click could theoretically come through. I don't know if Chrome still fires in bad performance cases.

There is also this article that mentions "no delay" and 300ms depending on meta tag: https://web.dev/mobile-touchandmouse/

But "no delay" also means at least 1ms in my test, probably an internal Chrome "setTimeout" with 0ms, but it takes some time anyway.

Reply:
I'm pretty happy to have 300ms and this working over it not working at all. 

We can always tweak the figure later on or rework it. Given I'm the only one that has spotted this so far, we shouldn't have a problem :)